### ISSUE TITLE
Financial Data Charts range selector not rendered

### ISSUE THREAD

**Issue by KaushikEDS** [IgniteUI/igniteui-angular#10767]
## Question  
I am providing data to Financial charts and I cant see the range when I load data from my API.
Real Time Chart with missing range selector:-
![image](https://user-images.githubusercontent.com/79090369/147474671-dfdd26a9-5e64-41c4-ae99-77268ef09583.png)

Real Time Chart data in console :-
![image](https://user-images.githubusercontent.com/79090369/147474801-c09d305b-9fa5-4a1d-8f3f-679fec75f0aa.png)

But when I load mock data it shows the range selector. 
Mock Data Chart :-

![image](https://user-images.githubusercontent.com/79090369/147473109-47744a9e-c6a8-4a27-858a-65205524a2a7.png)

You can see the data I pass to the component in the console.
![image](https://user-images.githubusercontent.com/79090369/147473201-61b09bf0-1065-4894-b757-1e4c2044ceb4.png)

Mock Data JSON link : https://paste.ofcode.org/WdHQqc22yUGyQ7CKMjmdPp

Real Data JSON link : https://paste.ofcode.org/ESaCXCkJH9EvguKuxi36VA

**Comment by amitmoondra** [comment]
Can this be reviewed early, we started using charts and got this issue.

**Comment by hanastasov** [comment]
Hi @amitmoondra and @KaushikEDS

Thanks for reaching out to us and the data shared. Actually, I get a result without range displayed using both data sources, and this is due to the fact that the `time` property holds a string value that is not recognized as date. Converting the date to real javascript date objects resolved the issue on my end.

At the same time, the data in your console shows date objects for the `time` values. IS there any chance that the console is showing a data collection which is not the data collection being fed to the chart?

To make sure what the dat fed to the chart is, do the following: in the DOM, select the `<igx-financial-chart>` element, and then run   `ng.getComponent($0)._dataSource` in the console:

![image](https://user-images.githubusercontent.com/14248932/147652834-084fba83-c742-49aa-95fb-1ba6b2265aab.png)

**Comment by KaushikEDS** [comment]
@hanastasov the chart renders in both cases (Mock and Api call).
I want to understand how the range selector shows up.

By range selector I mean these buttons:-
![image](https://user-images.githubusercontent.com/79090369/147653354-ea2f8160-999c-472d-9ca7-ccc0e385c9e2.png)
The screenshot you provided is also missing the Range selector.


I want to provide correct data so that it always shows up and if it doesn't show up I want to explain to my client the reason for that.

The data I provide to the chart has the time attribute as a date object, so don't worry it is the actual data I provided in the console which is fed to the chart. The reason for it to appear string is copy-pasting a date object converts it to a string.

**Comment by KaushikEDS** [comment]
@hanastasov I am able to see the Range selector I mentioned after I sorted the data according to the time.
I assumed the chart will plot it automatically without me needing to do so. 

**_Can you confirm if that is the solution to my problem?_** 
So I'll know for sure to provide sorted data.
I am thinking there are some issues with the X-axis of the chart. I have raised another issue mentioning the same at https://github.com/IgniteUI/igniteui-angular/issues/10770

**Comment by hanastasov** [comment]
@KaushikEDS yes, I have used a case to demonstrate that when `ng.getComponent($0)._dataSource` contains a string value for the `time` property, then the Range selector is not rendered.

Can you examine what is there in the `time` prop on your side if **before and after** sorting the data? 

I confirm this is a safe workaround for the issue.
